Xamarin.Essentials/Calendars/Calendars.uwp.cs

```diff
@@ -522,5 +522,67 @@ static async Task<bool> PlatformRemoveAttendeeFromEvent(CalendarEventAttendee ne
 
             return attendeeToRemove != null;
         }
+
+        static async Task<bool> PlatformAddReminderToEvent(CalendarEventReminder calendarEventReminder, string eventId)
+        {
+            await Permissions.RequestAsync<Permissions.CalendarWrite>();
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+
+            Appointment uwpAppointment;
+            try
+            {
+                uwpAppointment = await instance.GetAppointmentAsync(eventId);
+            }
+            catch (ArgumentException)
+            {
+                if (string.IsNullOrWhiteSpace(eventId))
+                {
+                    throw new ArgumentException($"[UWP]: No Event found for event Id {eventId}");
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException($"[UWP]: No Event found for event Id {eventId}");
+                }
+            }
+
+            uwpAppointment.Reminder = TimeSpan.FromMinutes(calendarEventReminder.MinutesPriorToEventStart);
+            var calendar = await instance.GetAppointmentCalendarAsync(uwpAppointment.CalendarId);
+            await calendar.SaveAppointmentAsync(uwpAppointment);
+            uwpAppointment = await instance.GetAppointmentAsync(eventId);
+
+            return uwpAppointment.Reminder != null;
+        }
+
+        static async Task<bool> PlatformReminderFromEvent(string eventId)
+        {
+            await Permissions.RequestAsync<Permissions.CalendarRead>();
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+
+            Appointment uwpAppointment;
+            try
+            {
+                uwpAppointment = await instance.GetAppointmentAsync(eventId);
+            }
+            catch (ArgumentException)
+            {
+                if (string.IsNullOrWhiteSpace(eventId))
+                {
+                    throw new ArgumentException($"[UWP]: No Event found for event Id {eventId}");
```

Comment:
Let's pass the native exception into the new Exceptions we're creating here

Reply:
Will be doing this in a separate PR after this one